src/data/cargo_dist.rs

```diff
@@ -0,0 +1,58 @@
+use axoasset::{Asset, LocalAsset};
+pub use cargo_dist_schema::{Artifact, ArtifactKind, DistManifest, Release};
+
+use crate::config::Config;
+use crate::data::github::GithubRelease;
+use crate::errors::*;
+
+#[derive(Clone)]
+pub struct DistRelease {
+    pub manifest: DistManifest,
+    pub source: GithubRelease,
+}
+
+pub fn get_os(name: &str) -> Option<&str> {
+    match name.trim() {
+        "x86_64-unknown-linux-gnu" => Some("linux"),
+        "x86_64-apple-darwin" => Some("mac"),
+        "aarch64-apple-darwin" => Some("arm mac"),
+        "x86_64-pc-windows-msvc" => Some("windows"),
+        &_ => None,
```

Comment:
this is definitely one of those functions i suspect i might move every refactor because i can't decide where it goes lol, def would be great to put in the cargo dist util